lldb/source/Host/posix/HostInfoPosix.cpp

```diff
@@ -139,8 +177,53 @@ FileSpec HostInfoPosix::GetDefaultShell() {
   return FileSpec("/bin/sh");
 }
 
+FileSpec HostInfoPosix::GetProgramFileSpec() {
+  static FileSpec g_program_filespec;
+
+  if (!g_program_filespec) {
+    char exe_path[PATH_MAX];
+    ssize_t len = readlink("/proc/self/exe", exe_path, sizeof(exe_path) - 1);
+    if (len > 0) {
+      exe_path[len] = 0;
+      g_program_filespec.SetFile(exe_path, FileSpec::Style::native);
+    }
+  }
+
+  return g_program_filespec;
+}
```

Comment:
I'm not too sure about this one. While other posix-like systems do have a "proc" file system, its contents tends to be very different. Juding by [this manpage](https://man.freebsd.org/cgi/man.cgi?query=procfs) the FreeBSD equivalent to `/proc/self/exe` *might* be `/proc/curproc/file`. @DavidSpickett, do you know if code like this would work for FreeBSD (with a change in the file name)?

If so, then I'd say this is sufficient reason to put this code here. Otherwise, I'd probably keep it in Linux. (Depending on how similar the two end up looking, we may want to have a common linux/aix class or accept the duplication)

Reply:
Let's leave this in Linux given that neither FreeBSD or NetBSD needed it until now.

I tried this on FreeBSD and I think this from the manpage is relevant:
```
This   functionality   is   deprecated. Users are  advised  to  use libprocstat(3) and kvm(3) instead.
```
I had to mount it manually and this is on an AWS instance so I expect any default FreeBSD install to be the same.

Once I did, only `/curproc/` worked, not `/self/`.

NetBSD allows `/self/` for Linux compatibility (https://man.netbsd.org/mount_procfs.8) but I wouldn't be surprised if it also didn't mount procfs by default.

Reply:
Just one concern with not moving `GetProgramFileSpec` , is that `ComputeSupportExeDirectory` relies on calling this, in case the file_spec is not found. So we'll have to restore `ComputeSupportExeDirectory` as well back to Linux, unless we have a better option.

Reply:
You should be able to call the function "virtually" as `HostInfo::GetProgramFileSpec()` even though it's implemented only in the derived class. All of the functions in `HostInfoBase` do that, so I don't see why that wouldn't work here.

Reply:
Okay got it. Resolved it.